### DISCUSSION TITLE
peer-has doesn't work as intended.

### DISCUSSION THREAD

**Discussion by JUSTIVE** [tailwindlabs/tailwindcss#13889] [Help]
thought that [this example](https://play.tailwindcss.com/LP8yWyNEJC) would work, the 'asdf' being colored when input, the sibling's child is focused, but not.

By flipping the sibling's order like [this](https://play.tailwindcss.com/LGgrstJcKN) worked as I intended, the order of peers matters. is this an intended spec? I wasn't able to find these caveats anywhere.

**Comment by wongjn** [comment]
Yes, this is an intended spec. As per [the documentation](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state):

> It’s important to note that the peer marker can only be used on previous siblings because of how the [subsequent-sibling combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Subsequent-sibling_combinator) works in CSS.
>  
> ❌ **Won’t work, only previous siblings can be marked as peers**  
> 

> ```html
> <label>
>   <span class="peer-invalid:text-red-500 ...">Email</span>
>   <input type="email" class="peer ..."/>
> </label>
> ```

For your provided case, you could use `:has()` without peer: https://play.tailwindcss.com/cjacs9O6er
